astropy/wcs/tests/test_wcs.py

```diff
@@ -1915,3 +1923,349 @@ def test_DistortionLookupTable():
             img_world_wcs.pixel_to_world_values(12 + dx * 3, 22 + dy * 3),
             [12 + dx * 3, 22 + dy * 3],
         )
+
+
+HEADER_WITH_NON_SI_UNITS = """
+WCSAXES = 5
+CTYPE1  = 'RA---TAN'
+CTYPE2  = 'FREQ'
+CTYPE3  = 'DEC--TAN'
+CTYPE4  = 'WAVE'
+CTYPE5  = 'OFFSET'
+CUNIT1  = 'arcsec'
+CUNIT2  = 'GHz'
+CUNIT3  = 'arcsec'
+CUNIT4  = 'nm'
+CUNIT5  = 'arcmin'
+CRVAL1  = 4
+CRVAL2  = 5
+CRVAL3  = 6
+CRVAL4  = 7
+CRVAL5  = 8
+CRPIX1  = 1
+CRPIX2  = 2
+CRPIX3  = 3
+CRPIX4  = 4
+CRPIX5  = 5
+CDELT1  = 4
+CDELT2  = 3
+CDELT3  = 2
+CDELT4  = 1
+CDELT5  = 6
+""".strip()
+
+
+class TestPreserveUnits:
+    def setup_method(self, method):
+        self.header = fits.Header.fromstring(HEADER_WITH_NON_SI_UNITS, sep="\n")
+        self.wcs_default = wcs.WCS(self.header)
+        self.wcs_preserve = wcs.WCS(self.header, preserve_units=True)
+        self.ones = np.ones((3, 5))
```

Comment:
I think `ones` is not a good description of what this array is (for). Maybe something like `ref_coordinates` or something like that. Would it also be possible to make this array with values other than 0 or 1? Maybe make it a larger array (but !=5, of course)?

Reply:
Done - using deterministic random values now